src/main/java/hudson/plugins/promoted_builds/Promotion.java

```diff
@@ -77,14 +77,30 @@ public Promotion(PromotionProcess project, File buildDir) throws IOException {
         super(project, buildDir);
     }
 
+
+    /**
+     * Gets the build that this promotion promoted.
+     * @since 3.4
+     * @return
+     *      {@code null} if there's no such object. For example, if the build has already garbage collected.
+     */
+    @Exported
```

Comment:
Duplicate export into API, remove.

Reply:
I am not sure about the compatibility impact, so I would rather keep the duplication for now. Later I plan to cut 4.0 and nuke half of this stuff anyway

Reply:
This is the _remote API_. You are going to introduce an unnecessary breaking change in the remote API in 4.0 if you keep this in and remove it then.

Reply:
Agreed. I will likely keep the method itself but just break the return type, but we can figure it out later
```suggestion
```

Reply:
Right, this thread is specifically about `@Exported`, I understand why the rest is there.